src/TagsInput/TagsInput.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+
+import * as React from 'react';
+import { View, Platform, TextInput } from 'react-native';
+import type { ViewLayoutEvent } from 'react-native/Libraries/Components/View/ViewPropTypes';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+
+import DeleteButton from './components/DeleteButton';
+import TagsContainer from './components/TagsContainer';
+import InputField from './components/InputField';
+import { StyleSheet } from '../PlatformStyleSheet';
+import { Text } from '../Text';
+
+import { MIN_WIDTH } from './constants';
+
+type Props = {|
+  +label: string,
+  +onChangeText: (value: string) => void,
+  +placeholder: string,
+  +selected?: string[],
+  +disabled?: boolean,
+  +fontSize: number,
+  +onClearPress?: () => void,
+  +value?: string,
+|};
+
+type State = {
+  value: string,
+  containerWidth: ?number,
+  inputWidth: ?number,
+};
+
+export default class TagInput extends React.Component<Props, State> {
+  inputRef: ?{ current: null | React.Element<typeof TextInput> };
+  static defaultProps = {
+    fontSize: 16,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.inputRef = React.createRef();
+
+    this.state = {
+      value: props.value ?? '',
+      containerWidth: null,
+      inputWidth: null,
+    };
+  }
+
+  componentDidUpdate = (prevProps: Props) => {
+    const { value } = this.props;
+    if (value !== prevProps.value) {
+      this.setState({ value });
+    }
+  };
+
+  setContainerWidth = (event: ViewLayoutEvent) => {
+    this.setState({ containerWidth: event.nativeEvent.layout.width });
+  };
+
+  setInputWidth = (event: ViewLayoutEvent) => {
+    this.setState({ inputWidth: event.nativeEvent.layout.width });
+  };
+
+  getWidth = (width: ?number) => {
+    const { containerWidth } = this.state;
+    if (containerWidth && width) {
+      const newWidth = containerWidth - width;
+      return newWidth < 0 ? 'auto' : newWidth;
+    }
+    return null;
+  };
+
+  getPlaceholder = () => {
+    const { selected, placeholder } = this.props;
+    return selected?.length === 0 && placeholder ? placeholder : null;
+  };
+
+  handleChange = (value: string) => {
+    const { value: oldValue } = this.state;
+    const { onChangeText } = this.props;
+    if (value !== oldValue) {
+      this.setState({ value });
+      onChangeText(value);
+    }
+  };
+
+  handleClear = () => {
+    const { onClearPress } = this.props;
+    this.setState({ value: '' });
+
+    // $FlowFixMe property focus is missing in object type
+    Platform.OS === 'web' && this.inputRef.current.focus();
+
+    if (onClearPress) {
```

Comment:
I believe you can even do `onClearPress?.()`.

Reply:
thanks, I have been learned something new.